hytra/core/hypothesesgraph.py

```diff
@@ -173,16 +173,21 @@ def checkNodeWhileAddingLinks(frame, obj):
             if (frame, obj) not in self._graph:
                 getLogger().warning("Adding node ({}, {}) when setting up links".format(frame, obj))
 
-        kdTreeNextFrame = None
+        kdTreeFrames = [None]*(skipLinks+1)
+        # kdTreeNextFrame = None
         for frame in range(len(probabilityGenerator.TraxelsPerFrame.keys()) - 1):
             if frame > 0:
-                kdTreeThisFrame = kdTreeNextFrame
+                del kdTreeFrames[0] # this is the current frame
+                if frame + skipLinks < len(probabilityGenerator.TraxelsPerFrame.keys()):
+                    kdTreeFrames.append(self._buildFrameKdTree(probabilityGenerator.TraxelsPerFrame[frame + skipLinks]))
+                    self._addNodesForFrame(frame + skipLinks, probabilityGenerator.TraxelsPerFrame[frame + skipLinks])
+                else:
+                    kdTreeFrames.append(self._buildFrameKdTree(probabilityGenerator.TraxelsPerFrame[2*frame - skipLinks - len(probabilityGenerator.TraxelsPerFrame.keys()) + 1]))
```

Comment:
please add a comment why this index makes sense :) Why is there an `else` block in the first place? If `frame+skipLinks > numFrames` then I'd expect we do not need to build another KD tree?